tests/test_config_manager.py

```
@@ -196,3 +196,570 @@ def test_none_overrides_not_serialized(self, tmp_path: Path) -> None:
+        data: dict[str, object] = {"key": "value"}
+        # Different strings, identical numeric meaning.
+        result = migrate_to_current(data, from_version="1.0", to_version="1.0.0")
+        assert result is data, (
+            "numerically-equivalent versions must short-circuit "
+            "(round-9 MIG-2 / coderabbit hgCI follow-up)"
+        )
+        # Symmetric direction.
+        result2 = migrate_to_current(data, from_version="1.0.0", to_version="1.0")
+        assert result2 is data
+
+    def test_migration_walker_stops_on_registry_gap(self, monkeypatch: pytest.MonkeyPatch) -> None:
```

Ruff (0.15.11)

[warning] Unused method argument: `monkeypatch`

(ARG002)